k2/csrc/pytorch_context.h

```diff
@@ -15,10 +13,85 @@
 #ifndef K2_CSRC_PYTORCH_CONTEXT_H_
 #define K2_CSRC_PYTORCH_CONTEXT_H_
 
+#include "c10/cuda/CUDACachingAllocator.h"
 #include "k2/csrc/context.h"
+#include "k2/csrc/log.h"
+#include "torch/torch.h"
+
+namespace k2 {
 
 class PytorchContext : public Context {
-  // ...
+ public:
```

Comment:
It would be more efficient to have separate classes for the GPU and CPU contexts, so we can avoid that if-statement.  Things like GetDeviceType() are called quite often so I think this is worthwhile.  But that's not a blocking change, can do it later.

Reply:
I will split it to two classes: `PytorchCpuContext` and `PytorchCudaContext`.

Reply:
Fixed in #163 .